testsuite/features/support/env.rb

```diff
@@ -150,7 +150,12 @@ def capybara_register_driver
   log "This scenario took: #{current_epoch - @scenario_start_time} seconds"
   if scenario.failed?
     begin
-      if web_session_is_active?
+      if scenario.exception.is_a?(Selenium::WebDriver::Error::WebDriverError)
+        log "Caught web driver error: #{scenario.exception.message}"
+        Capybara.current_session.driver.quit
+        visit Capybara.app_host
+        log 'Web driver has been restarted'
+      elsif web_session_is_active?
```

Comment:
Could it happen that w ehave a WebDriverError but the web session is active?
I'm not sure if we should order differently these if statements.

Reply:
Not according to what we saw.